IndexError on Matmul

When put vector parameters, matmul converter occurs an error.

```
    @support((1, 6, 7))
    def convert_MatMul(func, opset_version, input_names,
                       output_names, context, parameters):
        bias_shape = (
>           func.inputs[0].shape[-1] if func.transa else func.inputs[0].shape[-2],
            func.inputs[1].shape[-2] if func.transb else func.inputs[1].shape[-1]
        )
E       IndexError: Parameterized test failed.
E
E       Base test method: TestBinaryMathOperators.test_vector
E       Test parameters:
E         op_name: MatMul
E         ops: chainer.functions.matmul(a, b, transb=True)
E
E
E       (caused by)
E       IndexError: tuple index out of range
```

--- a/onnx_chainer/functions/math.py
+++ b/onnx_chainer/functions/math.py
@@ -136,22 +136,22 @@ def convert_Identity(func, opset_version, input_names,
     return onnx_helper.make_node('Identity', input_names, output_names),
 
 
-@support((1, 6, 7))
 def convert_MatMul(func, opset_version, input_names,
                    output_names, context, parameters):
-    bias_shape = (
-        func.inputs[0].shape[-1] if func.transa else func.inputs[0].shape[-2],
-        func.inputs[1].shape[-2] if func.transb else func.inputs[1].shape[-1]
-    )
-    bias_tensor = np.zeros(bias_shape, dtype=func.inputs[0].dtype)
-    bias_name = context.add_param(bias_tensor, 'bias')
-    input_names.append(bias_name)
+    ndim_a = len(func.inputs[0].shape)
+    ndim_b = len(func.inputs[1].shape)
 
-    return onnx_helper.make_node(
-        'Gemm', input_names, output_names,
-        transA=func.transa,
-        transB=func.transb
-    ),
+    gb = onnx_helper.GraphBuilder()
+    if ndim_a > 1 and func.transa:
+        perm = list(range(ndim_a))
+        perm[-1], perm[-2] = perm[-2], perm[-1]
+        input_names[0] = gb.op('Transpose', [input_names[0]], perm=perm)
+    if ndim_b > 1 and func.transb:
+        perm = list(range(ndim_b))
+        perm[-1], perm[-2] = perm[-2], perm[-1]
+        input_names[1] = gb.op('Transpose', [input_names[1]], perm=perm)
+    gb.op('MatMul', input_names)
+    return gb.nodes(output_names)
 
 
 @support((1, 6, 8))

--- a/tests/functions_tests/test_maths.py
+++ b/tests/functions_tests/test_maths.py
@@ -1,5 +1,6 @@
 import chainer
 from chainer import testing
+import numpy as np
 
 from onnx_chainer.testing import input_generator
 from tests.helper import ONNXModelTest
@@ -79,14 +80,16 @@ class TestUnaryMathOperators(ONNXModelTest):
     {'op_name': 'Sub', 'ops': 'a - b'},
     {'op_name': 'Mul', 'ops': 'a * b'},
     {'op_name': 'Div', 'ops': 'a / b'},
-    {'op_name': 'MatMul',
+    {'op_name': 'MatMul_transa',
+     'ops': 'chainer.functions.matmul(a, b, transa=True)'},
+    {'op_name': 'MatMul_transb',
      'ops': 'chainer.functions.matmul(a, b, transb=True)'},
     {'op_name': 'Maximum', 'ops': 'chainer.functions.maximum(a, b)'},
     {'op_name': 'Minimum', 'ops': 'chainer.functions.minimum(a, b)'},
 )
 class TestBinaryMathOperators(ONNXModelTest):
 
-    def setUp(self):
+    def get_model(self):
         class Model(chainer.Chain):
 
             def __init__(self, ops):
@@ -94,20 +97,34 @@ class TestBinaryMathOperators(ONNXModelTest):
                 self.ops = ops
 
             def __call__(self, a, b):
-                if not isinstance(a, chainer.Variable):
-                    a = chainer.Variable(a)
-                if not isinstance(b, chainer.Variable):
-                    b = chainer.Variable(b)
                 return eval(self.ops)
 
-        self.model = Model(self.ops)
-        a = chainer.Variable(input_generator.increasing(2, 3))
-        b = chainer.Variable(input_generator.nonzero_increasing(2, 3) * 0.3)
-        self.x = (a, b)
+        return Model(self.ops)
 
     def test_output(self):
-        name = self.op_name.lower()
-        self.expect(self.model, self.x, name=name)
+        test_cases = ['matrix', 'vector']
+        if not self.op_name.startswith('MatMul'):
+            test_cases.append('scalar')
+
+        for tc in test_cases:
+            xs = getattr(self, tc)()
+            name = '{}_{}'.format(self.op_name.lower(), tc)
+            self.expect(self.get_model(), xs, name=name)
+
+    def matrix(self):
+        a = chainer.Variable(input_generator.increasing(5, 2, 3))
+        b = chainer.Variable(input_generator.nonzero_increasing(5, 2, 3) * 0.3)
+        return (a, b)
+
+    def vector(self):
+        a = chainer.Variable(input_generator.increasing(2,))
+        b = chainer.Variable(input_generator.nonzero_increasing(2,) * 0.3)
+        return (a, b)
+
+    def scalar(self):
+        a = chainer.Variable(np.array(7, dtype=np.float32))
+        b = chainer.Variable(np.array(13, dtype=np.float32))
+        return (a, b)
 
 
 @testing.parameterize(

